data/src/main/kotlin/com/antosik/benchproject/data/popular/movies/repo/PopularMoviesDataRepository.kt

```diff
@@ -14,9 +17,30 @@ internal class PopularMoviesDataRepository(
 
     override suspend fun getPopularMoviesList(): List<MovieModel> {
         runCatching {
-            val remoteMovies = apiService.getPopularMovies(BuildConfig.API_KEY)
-            dao.insertMovies(remoteMovies.toEntity())
+            dao.insertMovies(fetchAndMergeMoviesWithLocalData())
         }
         return dao.getMovies().toDomain()
     }
+
+    override suspend fun updatePopularMovie(movieModel: MovieModel) {
+        runCatching {
```

Comment:
do we need runCatching, if updateMovie is already a suspend fun and we are actually not catching anything ?

Reply:
Yes, It is not needed here because I not catching anything, and also some problems shouldn't occur here. I will delete `runCatching`.

Reply:
Done 😄